### ISSUE TITLE
Calling samp.removeEventListener from within an event crashes the server

### ISSUE THREAD

**Issue by ElDuderino420** [AmyrAhmady/samp-node#12]
For some reason when I try to remove an event during a separate event trigger, the server crashes. It will crash regardless if any function references are provided.

```js
//will crash server
samp.on('OnPlayerConnect', function(playerid) {
    samp.removeEventListener('OnPlayerSpawn');
});
```
```js
//works as intended
samp.on('OnPlayerConnect', function(playerid) {
    console.log('onplayerconnect')
});

samp.removeEventListener('OnPlayerSpawn');
```

**Comment by AmyrAhmady** [comment]
Right now most of functionalities inside of an event listener's scope requires making your Promise handler, or when you are having an uncaught exception in there, otherwise you will receive a crash cause of bad thread management I have (basically none) in samp-node, this will be fixed in future but right now I am busy with another project. I'll return to this soon, so you can either wait or if you are capable of contributing, you can help :)

**Comment by AmyrAhmady** [comment]
Probably fixed by #15 , if it still happens, please report it again by creating a new issue

**Comment by plutalov** [comment]
> Probably fixed by #15 , if it still happens, please report it again by creating a new issue

Still crashes. I'm gonna fix it along with the NodeJS upgrade to v16

**Comment by AmyrAhmady** [comment]
Fixed by this https://github.com/AmyrAhmady/samp-node/commit/3961a068decac080ebe26f29140c69dc2e314d97
